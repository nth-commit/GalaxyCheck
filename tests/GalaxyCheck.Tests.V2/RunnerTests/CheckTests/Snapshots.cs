using GalaxyCheck;
using NebulaCheck;
using Snapshooter;
using Snapshooter.Xunit;
using System.Collections.Immutable;
using System.Linq;
using Xunit;


namespace Tests.V2.RunnerTests.CheckTests
{
    public class Snapshots
    {
        [Fact]
        public void Unfalsifiable() => RunSnapshots(GalaxyCheck.Gen.Int32().ForAll((x) => true));

        [Fact]
        public void Unfalsifiable_Filtered() => RunSnapshots(GalaxyCheck.Gen.Int32().Where(x => x % 2 == 0).ForAll((x) => true));

        [Fact]
        public void Falsifiable_AtLargerSizes() => RunSnapshots(GalaxyCheck.Gen.Create(parameters => (parameters.Size.Value, parameters)).ForAll(x => x < 50));

        [Fact]
        public void Falsifiable_AtSmallerSizes() => RunSnapshots(GalaxyCheck.Gen.Create(parameters => (parameters.Size.Value, parameters)).ForAll(x => x > 50));

        private static void RunSnapshots(GalaxyCheck.Property<int> property)
        {
            var seed = 0;

            foreach (var iterations in new[] { 1, 2, 50, 100 })
            {
                var check = property.Check(seed: seed, iterations: iterations);

                Snapshot.Match(
                    check with { Checks = ImmutableList.Create<GalaxyCheck.Runners.Check.CheckIteration<int>>() },
                    SnapshotNameExtension.Create($"Iterations={iterations}"));
            }
        }
    }
}
