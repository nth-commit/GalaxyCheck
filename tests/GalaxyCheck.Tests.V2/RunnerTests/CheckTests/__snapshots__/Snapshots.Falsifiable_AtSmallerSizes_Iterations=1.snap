{
  "Iterations": 1,
  "Discards": 0,
  "Shrinks": 0,
  "Counterexample": null,
  "Checks": [],
  "InitialParameters": {
    "Rng": {
      "Family": 0,
      "Seed": 0,
      "Order": 0
    },
    "Size": {
      "Value": 100
    }
  },
  "NextParameters": {
    "Rng": {
      "Family": 0,
      "Seed": 0,
      "Order": 0
    },
    "Size": {
      "Value": 0
    }
  },
  "TerminationReason": "ReachedMaximumIterations",
  "Falsified": false,
  "RandomnessConsumption": 0
}
